### PR TITLE
Implement double complex type

### LINKED ISSUE #871: [SciPy]: Type ```DOUBLE COMPLEX``` not implemented yet
```
semantic error: Type not implemented yet.
    --> /home/pranavchiku/scipy/scipy/special/specfun/specfun.f:5298:9
     |
5298 |         DOUBLE COMPLEX Z, CEI
     |         ^^^^^^^^^^^^^^^^^^^^^ 
```

### PR DESCRIPTION
**Pull request by Smit-create**
Fixes #871

### CODE REVIEW

**Comment by konradha** [comment]
Nice, double complex seems well-supported in the frontend with this.

**Comment by Pranavchiku** [comment]
Should we add the part of code present in scipy as a test? 
I tried to add but ig, I don't have access 

```fortran
        SUBROUTINE EIXZ(Z,CEI)
        IMPLICIT NONE
        DOUBLE COMPLEX Z, CEI
        DOUBLE PRECISION PI
        PI=3.141592653589793D0
        CALL E1Z(-Z, CEI)
        CEI = -CEI
        IF (DIMAG(Z)>0) THEN
           CEI = CEI + (0d0,1d0)*PI
        ELSE IF (DIMAG(Z)<0) THEN
           CEI = CEI - (0d0,1d0)*PI
		END IF
        RETURN
        END
```

File: integration_tests/complex_07.f90 [hunk 1]
```diff
@@ -5,10 +5,18 @@ program complex_07
...
+    print *, c1
+    x2 = dcmplx(real_part)
+    y2 = dcmplx(img_part)
+    z2 = dcmplx(real_part, img_part)
     if (abs(x - (real_part, 0)) > 1e-5) error stop
     if (abs(y - (img_part, 0)) > 1e-5) error stop
     if (abs(z - (real_part, img_part)) > 1e-5) error stop
+    if (abs(x2 - (real_part, 0)) > 1e-5) error stop
+    if (abs(y2 - (img_part, 0)) > 1e-5) error stop
+    if (abs(z2 - (real_part, img_part)) > 1e-5) error stop
```
Comment (certik) [inline integration_tests/complex_07.f90:19]:
```suggestion
    if (abs(x2 - (real_part, 0)) > 1e-10_dp) error stop
    if (abs(y2 - (img_part, 0)) > 1e-10_dp) error stop
    if (abs(z2 - (real_part, img_part)) > 1e-10_dp) error stop
```
And define `dp` as `integer, parameter :: dp = kind(0.d0)`

**Review by certik — APPROVED** [review]
Left some minor comments that can be fixed in subsequent PRs. Thanks!

**Pull request merged**